pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -324,3 +343,29 @@ func collectVMIInterfaceInfo(vmi *k6tv1.VirtualMachineInstance, iface k6tv1.Virt
 		Value: 1.0,
 	}
 }
+
+func reportMigrationTimestamps(vmi *k6tv1.VirtualMachineInstance) []operatormetrics.CollectorResult {
+	var cr []operatormetrics.CollectorResult
+
+	if vmi.Status.MigrationState == nil {
+		return cr
+	}
+
+	if vmi.Status.MigrationState.StartTimestamp != nil {
```

Comment:
why not query the migration object (virtualmachineinstancemigrations ) directly? so the metric will show for sure catch the migration. 

when I used `kubectl describe vmim vm-cirros-migration` I see:
```
Name:         vm-cirros-migration2
Namespace:    default
Labels:       kubevirt.io/vmi-name=vm-cirros
Annotations:  kubevirt.io/latest-observed-api-version: v1
              kubevirt.io/storage-observed-api-version: v1
API Version:  kubevirt.io/v1
Kind:         VirtualMachineInstanceMigration
Metadata:
  Creation Timestamp:  2024-12-08T19:29:06Z
  Generation:          1
  Resource Version:    2604277
  UID:                 92ca7de8-da91-4215-874a-bc8d1b7ba9b0
Spec:
  Vmi Name:  vm-cirros
Status:
  Migration State:
    Completed:      true
    End Timestamp:  2024-12-08T19:29:14Z
    Migration Configuration:
      Allow Auto Converge:                    false
      Allow Post Copy:                        false
      Bandwidth Per Migration:                0
      Completion Timeout Per Gi B:            150
      Node Drain Taint Key:                   kubevirt.io/drain
      Parallel Migrations Per Cluster:        5
      Parallel Outbound Migrations Per Node:  2
      Progress Timeout:                       150
      Unsafe Migration Override:              false
    Migration UID:                            92ca7de8-da91-4215-874a-bc8d1b7ba9b0
...
    Source Pod:                               virt-launcher-vm-cirros-r29hw
    Start Timestamp:                          2024-12-08T19:29:11Z
....
    Target Pod:                          virt-launcher-vm-cirros-kp99d
  Phase:                                 Succeeded
  Phase Transition Timestamps:
    Phase:                       Pending
    Phase Transition Timestamp:  2024-12-08T19:29:06Z
    Phase:                       Scheduling
    Phase Transition Timestamp:  2024-12-08T19:29:06Z
    Phase:                       Scheduled
    Phase Transition Timestamp:  2024-12-08T19:29:11Z
    Phase:                       PreparingTarget
    Phase Transition Timestamp:  2024-12-08T19:29:11Z
    Phase:                       TargetReady
    Phase Transition Timestamp:  2024-12-08T19:29:11Z
    Phase:                       Running
    Phase Transition Timestamp:  2024-12-08T19:29:11Z
    Phase:                       Succeeded
    Phase Transition Timestamp:  2024-12-08T19:29:14Z
```

so we have the start+end time, the vm name, both pod names
*maybe it's the same as what you do now not sure

Reply:
Yes it exists after the migration ended- from what I saw at least 3 hours. If you delete the vmi it will be deleted as well which make sense

Reply:
So, it seems like a more stable place to get the info from.

Reply:
besides that, I saw that sometimes the VMIM took longer to get values populated

Reply:
> besides that, I saw that sometimes the VMIM took longer to get values populated
interesting, didn't encounter this- you think it's problematic?

Reply:
also I think maybe add the pods names also as labels? source_pod and target_pod?

Reply:
I think that info should be in other migration metrics, no?